### PR TITLE
Split UnitTests project into separate test projects

### LINKED ISSUE #857: Split unit tests into one test project per project under test
At the moment we're using one monolithic test project for all project in our solution. This means that to execute a single test, we need to build our entire solution. On my laptop it can easily take over a minute to execute a single test! This isn't very conducive to using testing. 😉 

I propose taking a light touch approach to this.

1. Duplicate the `UnitTests` project and rename/re-GUID for a project under test.
2. Add as Link source references to the tests for the target project.
3. Add as Link source references to any required test helper classes.
4. Remove any unused project, assembly and NuGet references.
5. Execute our unit tests quickly and rejoice. 🎈 

If/when we're happy with the new structure, we can consider moving our test source files into the individual test projects.

Does that make sense?

### LINKED ISSUE #1424: Remove all usages of Fody
Fody was used with XUnit for assembly/namespace level setup. Now that we are using NUnit, we no longer need to use Fody.

It looks like there are only a few usages for Fody in the Visual Studio project: https://github.com/github/VisualStudio/search?utf8=%E2%9C%93&q=Fody&type=

### LINKED ISSUE #1732: Make sure we're doing Splat.ModeDetector.Current.SetInUnitTestRunner(true) once per test project
Some of our tests appear to be failing depending on the order in which they're run. This is because `SetInUnitTestRunner(true)` isn't always getting called before a test is executed.

Since we're no longer using Fody, we should add the following to the `AssemblyInfo.cs` for each of our test projects:

```cs
[SetUpFixture]
public class SplatModeDetectorSetUp
{
    [OneTimeSetUp]
    public void RunBeforeAnyTests()
    {
        Splat.ModeDetector.Current.SetInUnitTestRunner(true);
    }
}
```

We should also remove `IEntryExitDecorator` and any other calls to `SetInUnitTestRunner(true)` in our test projects.

@meaghanlewis It might make sense to do this as part of #1721?

### PR DESCRIPTION
**Pull request by meaghanlewis**
Split out UnitTests and separate test projects.

Closes #857 
Closes #1424 
Closes #1732

### CODE REVIEW

File: test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
+
+[SetUpFixture]
+public class MySetUpClass
+{
+    [OneTimeSetUp]
+    public void RunBeforeAnyTests()
```
Comment (StanleyGoldman) [inline test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs]: asdf

File: test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible 
+// to COM components.  If you need to access a type in this assembly from 
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("3525d819-6aec-4879-89fb-56b41f026571")]
```
Comment (StanleyGoldman) [inline test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs]: asdf

Comment (StanleyGoldman) [inline test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs]: asdf

Comment (StanleyGoldman) [inline test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs]: asdf

File: test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
+
+[SetUpFixture]
+public class MySetUpClass
+{
+    [OneTimeSetUp]
+    public void RunBeforeAnyTests()
+    {
+        Splat.ModeDetector.Current.SetInUnitTestRunner(true);
```
Comment (StanleyGoldman) [inline test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs]: asdf

Comment (StanleyGoldman) [inline test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs]: Asdf

File: test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+[assembly: Guid("3525d819-6aec-4879-89fb-56b41f026571")]
+
+// Version information for an assembly consists of the following four values:
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
```
Comment (StanleyGoldman) [inline test/GitHub.App.UnitTests/Properties/AssemblyInfo.cs]: asdf

**Comment by grokys** [comment]
Thanks @jcansdale - that hack has improved tests under ncrunch. There are still some failures but there always were. I think we should try updating RxUI/Splat at some point to see if things get fixed.

I've merged master into this branch now.

**Pull request merged**